hack/fake-apiserver/go.mod

```diff
@@ -9,8 +9,8 @@ require (
 	k8s.io/api v0.31.7
 	k8s.io/apimachinery v0.31.7
 	k8s.io/client-go v0.31.7
-	sigs.k8s.io/cluster-api v1.9.5
```

Comment:
/hold

FKAS needs bump to CAPI 1.10 as well.
@adilGhaffarDev  ?

Reply:
there are some deprecations in 1.10 that are affecting FKAS, there is an open issue for that: https://github.com/metal3-io/cluster-api-provider-metal3/issues/2378 
First, we have to fix these issues, let's do it in the next release cycle.

Reply:
Let's not hold the rest of CAPM3 because of FKAS

Reply:
OK, its fine if there is actual issues with it. Even better there is an issue for them. Next release is fine.
/unhold